### PR TITLE
KAA-1224: Incorrect calculation of log delivery time

### PR DESCRIPTION
**Pull request by mlysenko**

### CODE REVIEW

File: client/client-multi/client-cpp/kaa/channel/transport/ILoggingTransport.hpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ class ILoggingTransport {
      * @see LogSyncResponse
      *
      */
-    virtual void onLogSyncResponse(const LogSyncResponse& response) = 0;
+    virtual void onLogSyncResponse(const LogSyncResponse& response, std::size_t deliveryTime) = 0;
```
Comment (ohamov) [inline client/client-multi/client-cpp/kaa/channel/transport/ILoggingTransport.hpp:46]: document `deliveryTime` parameter.

File: client/client-multi/client-cpp/kaa/log/ILogProcessor.hpp [hunk 2]
```diff
@@ -34,7 +34,7 @@ class ILogProcessor {
      *
      * \param   response    Response from operations service.
      */
-    virtual void onLogUploadResponse(const LogSyncResponse& response) = 0;
+    virtual void onLogUploadResponse(const LogSyncResponse& response, std::size_t deliveryTime) = 0;
```
Comment (ohamov) [inline client/client-multi/client-cpp/kaa/log/ILogProcessor.hpp:38]: the same

File: client/client-multi/client-cpp/test/impl/log/LogCollectorTest.cpp [hunk 3]
```diff
@@ -261,7 +261,7 @@ BOOST_AUTO_TEST_CASE(SuccessDeliveryTest)
     deliveryStatus.result = SyncResponseResultType::SUCCESS;
     response.deliveryStatuses.set_array({ deliveryStatus });
 
-    logCollector.onLogUploadResponse(response);
+    logCollector.onLogUploadResponse(response, 1);
```
Comment (ohamov) [inline client/client-multi/client-cpp/test/impl/log/LogCollectorTest.cpp]: no magic numbers

**Review by ohamov — CHANGES_REQUESTED** [review]
Reviewed

**Comment by vadimol** [comment]
@kaa-leeroy retest this please

File: client/client-multi/client-cpp/test/impl/log/LogCollectorTest.cpp [hunk 4]
```diff
@@ -41,6 +41,8 @@
 
 namespace kaa {
 
+const std::size_t logDeliveryTime = 1;
```
Comment (ohamov) [inline client/client-multi/client-cpp/test/impl/log/LogCollectorTest.cpp]: I'd rename it to point out that this variable is nothing but the mock and doesn't affect the tests results.

File: client/client-multi/client-cpp/kaa/channel/transport/ILoggingTransport.hpp [hunk 5]
```diff
@@ -41,9 +41,9 @@ class ILoggingTransport {
      *
      * @param response the response from the server
      * @see LogSyncResponse
-     *
+     * @param deliveryTime a time when the log was delivered
```
Comment (ohamov) [inline client/client-multi/client-cpp/kaa/channel/transport/ILoggingTransport.hpp]: `s/a time when the log was delivered/The time point the log was delivered at`

File: client/client-multi/client-cpp/kaa/log/ILogProcessor.hpp [hunk 6]
```diff
@@ -32,9 +32,10 @@ class ILogProcessor {
     /**
      * Called when log upload response arrived.
      *
-     * \param   response    Response from operations service.
+     * \param   response        Response from operations service.
+     * \param   deliveryTime    Time when the log was delivered.
```
Comment (ohamov) [inline client/client-multi/client-cpp/kaa/log/ILogProcessor.hpp]: `s/Time when the log was delivered/The time point the log was delivered at`

**Review by ohamov — CHANGES_REQUESTED** [review]

**Comment by mlysenko** [comment]
The PR has been updated. Please review.

**Comment by vadimol** [comment]
@kaa-leeroy retest this please

**Review by ohamov — APPROVED** [review]
LGTM

**Review by vadimol — APPROVED** [review]
LGTM

**Pull request merged**